pkg/checks/external/ssl_util/main.go

```diff
@@ -0,0 +1,100 @@
+//Package ssl_util implements ssl check functions to kuberhealthy.
+//These test TLS connectivity to a host and check current expiration
+//status and time until certificate expiration.
+
+package ssl_util
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"net"
+	"strconv"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+var TimeoutSeconds = 10
+
+// CertHandshake performs a basic TLS/SSL handshake on the specified host and port, returning the cert information and any errors
+func CertHandshake(host, port string) error {
+	log.Info("Testing SSL handshake on host ", host, " over port ", port)
+
+	d := &net.Dialer{
+		Timeout: time.Duration(TimeoutSeconds) * time.Second,
+	}
+
+	conn, err := tls.DialWithDialer(d, "tcp", host+":"+port, &tls.Config{
+		InsecureSkipVerify: false,
+		MinVersion:         tls.VersionTLS12,
+	})
+
+	if err != nil {
+		log.Warnln([]*x509.Certificate{&x509.Certificate{}}, "", err)
+		return err
+	}
+	defer conn.Close()
+
+	err = conn.Handshake()
+	if err != nil {
+		log.Warn("Unable to complete SSL handshake with host", host+":", err)
+		return err
+	}
+	log.Info("SSL handshake to host", host, "completed successfully")
+	return err
+}
+
+// CertExpiry checks that the cert on a given host and port is not currently expired, and that the expiration day is the specified number of days away
+func CertExpiry(host, port, days string) (bool, bool, error) {
+	log.Info("Testing SSL expiration on host ", host, " over port ", port)
+	var certExpired bool
+	var expireWarning bool
+	var currentTime time.Time
+	var daysUntilInvalid uint
+	var cert []*x509.Certificate
```

Comment:
i dont think these 3 vars need to be instantiated here: `currentTime`, `daysUntilInvalid`, `cert`